### ISSUE TITLE
Leaf value

### ISSUE THREAD

**Issue by FedorTarasow** [catboost/catboost#2267]
Problem:
Leaf value is calculated differently when l2_leaf_reg equals default value (3) and fixed value (3). Should they be all equal? Can you please explain how leaf value is calculated?
![Screenshot_2](https://user-images.githubusercontent.com/122746903/212560203-6ae26b53-f180-4285-8aa3-eaf3e82c2b54.png)
catboost version: 1.1.1
Operating System: Windows 10

Example:
`df=pd.DataFrame(
    {'Age':[12,87,44,19,32,14],
    'Target':[1,1,0,0,1,1]})
model=CatBoostClassifier(iterations=1,depth=1,random_state=0).fit(df['Age'],df['Target'])
model.plot_tree(tree_idx=0)
model.get_all_params()['l2_leaf_reg']
model=CatBoostClassifier(iterations=1,depth=1,random_state=0,l2_leaf_reg=3).fit(df['Age'],df['Target'])
model.plot_tree(tree_idx=0)
`

**Comment by Evgueni-Petrov-aka-espetrov** [comment]
the classifiers differ, because explicit l2_leaf_reg disables automatic learning rate selection
to get the same models, just set the same learning rate
```python
import catboost
import pandas as pd

df=pd.DataFrame( {'Age':[12,87,44,19,32,14], 'Target':[1,1,0,0,1,1]})
model=catboost.CatBoostClassifier(iterations=1,depth=1,random_state=0).fit(df['Age'],df['Target'])
model.plot_tree(tree_idx=0)
print(model.get_all_params())
model=catboost.CatBoostClassifier(iterations=1,depth=1,random_state=0,l2_leaf_reg=3,learning_rate=0.5).fit(df['Age'],df['Target'])
print(model.get_all_params())
model.plot_tree(tree_idx=0)
```

**Comment by ccylance** [comment]
> the classifiers differ, because explicit l2_leaf_reg disables automatic learning rate selection to get the same models, just set the same learning rate -- see below
> 
> ```python
> import catboost
> import pandas as pd
> 
> df=pd.DataFrame( {'Age':[12,87,44,19,32,14], 'Target':[1,1,0,0,1,1]})
> model=catboost.CatBoostClassifier(iterations=1,depth=1,random_state=0).fit(df['Age'],df['Target'])
> model.plot_tree(tree_idx=0)
> print(model.get_all_params())
> model=catboost.CatBoostClassifier(iterations=1,depth=1,random_state=0,l2_leaf_reg=3,learning_rate=0.5).fit(df['Age'],df['Target'])
> print(model.get_all_params())
> model.plot_tree(tree_idx=0)
> ```

By the way, could you explain how the leaf value calculated? Im feeling confused about this because the paper does not demonstrate the details.
